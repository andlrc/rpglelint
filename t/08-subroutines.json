[
   {
      "column":7,
      "file":"t/08-subroutines.rpgle",
      "line":6,
      "message":"subroutine 'my_sub' is not allowed",
      "severity":"warning"
   },
   {
      "column":9,
      "file":"t/08-subroutines.rpgle",
      "line":18,
      "message":"subroutine 'my_sub' is not allowed",
      "severity":"warning"
   },
   {
      "column":9,
      "file":"t/08-subroutines.rpgle",
      "line":22,
      "message":"subroutine 'my_sub2' is not allowed",
      "severity":"warning"
   }
]
