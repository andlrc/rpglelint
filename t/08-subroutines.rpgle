// rpglelint: -Wsubroutine
dcl-s glob int(10);

exsr my_sub;

begsr my_sub;
  glob += my_proc();
endsr;

dcl-proc my_proc;
  dcl-pi *n varchar(32);
  end-pi;
  dcl-s ret varchar(32);

  exsr my_sub;
  return ret;

  begsr my_sub;
    exsr my_sub2;
  endsr;

  begsr my_sub2;
    ret = '32';
  endsr;
end-proc;
