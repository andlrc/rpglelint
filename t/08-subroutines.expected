t/08-subroutines.rpgle:6:7: warning: subroutine 'my_sub' is not allowed [-Wsubroutine]
 begsr my_sub;
       ^
t/08-subroutines.rpgle:18:9: warning: subroutine 'my_sub' is not allowed [-Wsubroutine]
   begsr my_sub;
         ^
t/08-subroutines.rpgle:22:9: warning: subroutine 'my_sub2' is not allowed [-Wsubroutine]
   begsr my_sub2;
         ^
